File: CodeMonk/diagonal_difference.cpp
#include <bits/stdc++.h>

using namespace std;

// Complete the diagonalDifference function below.
int diagonalDifference(vector<vector<int>> arr) {
    int first_diagonal_sum = 0;
    int sec_diagonal_sum = 0;
    int sum = 0;
    int n = arr.size();

    for (int i=0; i<arr.size();i++){
        for(int j=0; j<arr.size();j++){
            if (i==j){
                first_diagonal_sum= first_diagonal_sum + arr[i][j];
                }
        }
    }

    for(int i=0; i<=arr.size();i++){
        for(int j=0; j<=arr[0].size();j++){
            if (i==n-1-j){
                sec_diagonal_sum+=arr[i][j];
            }
        }
    }
    printf("%d\n", first_diagonal_sum);
    printf("%d", sec_diagonal_sum);
    return abs(first_diagonal_sum-sec_diagonal_sum);
}

int main()
{
    ofstream fout(getenv("OUTPUT_PATH"));

    int n;
    cin >> n;
    cin.ignore(numeric_limits<streamsize>::max(), '\n');

    vector<vector<int>> arr(n);
    for (int i = 0; i < n; i++) {
        arr[i].resize(n);

        for (int j = 0; j < n; j++) {
            cin >> arr[i][j];
        }

        cin.ignore(numeric_limits<streamsize>::max(), '\n');
    }

    int result = diagonalDifference(arr);

    fout << result << "\n";

    fout.close();

    return 0;
}

File: CodeMonk/diagonal_difference.py
#!/bin/python3

import math
import os
import random
import re
import sys

# Complete the diagonalDifference function below.
def diagonalDifference(arr):
    first_diagonal_sum = 0
    second_diagonal_sum = 0
    for i in range(len(arr)):
        for j in range(len(arr[0])):
            if i==j:
                first_diagonal_sum += arr[i][j]
    for i in range(len(arr)):
        for j in range(len(arr[0])):
            if i == n-1-j:
                second_diagonal_sum+=arr[i][j]

    diff = abs(first_diagonal_sum-second_diagonal_sum)
    return diff

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    n = int(input())

    arr = []

    for _ in range(n):
        arr.append(list(map(int, input().rstrip().split())))

    result = diagonalDifference(arr)

    fptr.write(str(result) + '\n')

    fptr.close()
